### DISCUSSION TITLE
Render URIs (ms-settings, ms-windows-store, etc.) as links

### DISCUSSION THREAD

**Discussion by aisgbnok** [community/community#16355] [Other Feature Feedback, Questions, & Ideas]
It would be great to accept URIs like `ms-settings` as links. Currently, if I enter a link such as `ms-settings:about` and I format it as a link using markdown syntax it doesn't work. `[ms-settings:about](ms-settings:about)` However if you copy and paste it into your browser and you are on a Windows machine it will launch the Windows Settings app and directly take you to the About page. 

Using a HTML formatted link doesn't work either `<a href="ms-windows-store://pdp/?ProductId=9N0DX20HK701">Windows Store: Terminal</a>` but copying and pasting `ms-windows-store://pdp/?ProductId=9N0DX20HK701` into your browser will directly open the Microsoft Store on Windows machines. 

[Here is Microsoft's URI documentation.](https://docs.microsoft.com/windows/uwp/launch-resume/launch-app-with-uri) Adding support for these links would be beneficial for support documentation, troubleshooting, etc. For example, adding URI links for your specific app, or maybe a specific Windows Setting in your repo's readme.

*[For a more detailed explanation, see my additional comment to a thread.](https://github.com/orgs/community/discussions/16355#discussioncomment-4802544)*

**Comment by mrherndon** [comment]
I was just looking for a way of doing this...  Where do we upvote!

**Reply by aisgbnok** [reply]
Unfortunately, all you can do at this time is click the up arrow. Additionally, add an emoji reaction if you wish.

![image](https://user-images.githubusercontent.com/10838153/215248463-689b35f9-6d25-4a1a-a09c-714c8fb2a90c.png)

Thanks for the support. I would have assumed given GitHub is a Microsoft company that this would have possibly been pushed down the pipeline by now as their staff might appreciate this feature.

**Comment by radar-blip** [comment]
Not sure if this is what you're looking for ?

`![ms-settings:about](ms-settings:about)` gets you this:  ![ms-settings:about](ms-settings:about)

**Reply by mrherndon** [reply]
That just creates a broken image with a link to the page that you are currently at

**Reply by aisgbnok** [reply]
**Thank you for your attempt and response,** I'm not sure what you did to try and create that link, but unfortunately it does not produce the result I am looking for. Though I am interested in what the raw markdown that produces that odd image link is? It's interesting because it opens in a new tab.

As for what I want, I thought my original post was clear:

> Currently, if I enter a link such as `ms-settings:about` and I format it as a link using markdown syntax it doesn't work. `[ms-settings:about](ms-settings:about)` However if you copy and paste it into your browser and you are on a Windows machine it will launch the Windows Settings app and directly take you to the About page.

Essentially I want GitHub markdown to recognize and correctly execute Microsoft URIs. On Windows (I believe 10 and up), you can paste a URI such as `ms-settings:bluetooth` into a modern web browser and it will trigger a URI event on the system. Windows will then essentially present the browser with the app that URI links to so it can ask the user if they want to open it.

![image](https://user-images.githubusercontent.com/10838153/215247857-9c124866-a8b8-4973-b659-b38f5544a489.png)

Once a user accepts the browser alert, the Windows app will be opened and the respective page will be navigated to.

![image](https://user-images.githubusercontent.com/10838153/215247905-a880d6f1-9c7c-4c1d-b2b2-81303a082848.png)

In this instance, the Windows Settings application opened and directly navigated to the Bluetooth page.

This is great for troubleshooting or linking individuals to system apps. As an example, see Microsoft's documentation on [Apply a provisioning package](https://learn.microsoft.com/windows/configuration/provisioning-packages/provisioning-apply-package). The first step has a `Recovery` link that uses a URI to take the user directly into the system settings.

![image](https://user-images.githubusercontent.com/10838153/215248023-2b8ed9bb-d0b1-4b0f-ac0b-4afb05cbc112.png)

I believe this can also be done for other apps and developers that use URIs in their own Windows applications. So having a quick link to open their application from say a README would be helpful and just provide a slightly more refined UX experience.

As a side note, some IDEs have this ability already. Like JetBrains. Although that is probably not a custom implementation from JetBrains and just the built in browser automatically executing the URI correctly. *Although I could be incorrect, not sure on the specifics.*

*I will link this comment to the main post for those looking for a more detailed explanation.*

**Reply by mrherndon** [reply]
I was trying to create a guide for my team on setting up our dev environment.  One of the requirements is to install WSL from the Microsoft store.  
I can link to the webpage with markup just fine. https://apps.microsoft.com/store/detail/windows-subsystem-for-linux/9P9TQF7MRM4R?hl=en-us&gl=us
But, they have to hit another button on that page to open the store on their device.  I was trying to save a step with the link directly to the ms-windows-store uri.  ms-windows-store://pdp/?ProductId=9P9TQF7MRM4R
I can live with the results, but it's weird that this doesn't work.  you can enter that uri into the chrome search bar and it works right out the gate, just like your settings link.  Which makes me feel that GitHub has intentionally prevented this from working.

**Reply by aisgbnok** [reply]
@mrherndon, yeah GitHub must use some some sort of pattern matching technique (probably with a whitelist component) that doesn't account for URIs.

---

As for your use case, just suggesting installing through PowerShell prompt
```PowerShell
wsl --install
```
*might* be better or more streamlined because [that command now is identical to pulling from the Microsoft store.](https://devblogs.microsoft.com/commandline/the-windows-subsystem-for-linux-in-the-microsoft-store-is-now-generally-available-on-windows-10-and-11/)

Depending on what distribution you want you can also do:
```PowerShell
wsl --install -d Ubuntu
```
```PowerShell
wsl --install -d kali-linux
```
or no distro
```PowerShell
wsl --install --no-distribution
```

That might provide a slightly more streamlined installation process than using the store anyway, but I don't know your exact workflow.

Just make sure the devices are updated. Build needs to be `190xx.2311` or higher on W10, or `22000.1281` or higher on W11.

> You will need to be running Windows 10 version 21H1, 21H2, or 22H2, or on Windows 11 21H2 with all of the November updates applied. You will know you have this update when you check that [KB5020030](https://support.microsoft.com/en-us/topic/november-15-2022-kb5020030-os-builds-19042-2311-19043-2311-19044-2311-and-19045-2311-preview-237a9048-f853-4e29-a3a2-62efdbea95e2) is installed on Windows 10, or [KB5019157](https://support.microsoft.com/en-us/topic/november-15-2022-kb5019157-os-build-22000-1281-preview-d64fb317-3435-49ff-b2c4-d0356a51a6b0) on Windows 11.
